server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -3267,6 +3302,13 @@ private boolean isRemoteStoreEnabled() {
         return (remoteStore != null && shardRouting.primary());
     }
 
+    public boolean isRemoteTxlogEnabledOnPrimary() {
+        return indexSettings != null
+            && indexSettings.isSegRepEnabled()
+            && indexSettings.isRemoteStoreEnabled()
+            && indexSettings.isRemoteTranslogStoreEnabled();
+    }
+
```

Comment:
We don't need all these checks since enabling remote translog is only possible if the other two are already enabled.  Can we avoid this and use AbstractIndexShardComponent#indexSettings()? Also let's use translog and avoid `txlog`.

Reply:
Done.